### DISCUSSION TITLE
Attaching Signature to Email

### DISCUSSION THREAD

**Discussion by norman262** [szimek/signature_pad#631] [Q&A]
I have successfully used **signature_pad** to capture signatures and save them to my database as a JSON object containing the points array and I can even view said signature in view-only mode by loading the said data. I do however need to attach said signatures to emails.

 **_Is there a way to generate an image of the signature server-side from the points data_**, or is there a better way to do this?

**Comment by UziTech** [comment]
You can use signature pad with node-canvas and call toDataURL. Or on the client side upload the image somewhere and save the URL.

**Comment by Rajan-Gupta1** [comment]
@norman262 Hello Norman, I have a similar use case to store the signature as a json and then render it in a web page. Can you please share the code snippet on what functions you called to create a json out of the signature and were able to render it in email?

**Reply by norman262** [reply]
Actually you cannot render it in an email from JSON points. The only way possible is to call the PNG method also and additionally store that in your database. Technically you should be able to embed this image data directly in your email but email clients such as Gmail consider this a risk and block it. Your best option therefore is to save an actual image file from your PNG data in a server folder and attach that to your outgoing email. I know, this is very disappointing and defeats the purpose of storing your signature in a light JSON format if you have to save the raw image anyway. For this reason, I have decided to simply add a "Signed by {person_name}" and a "View Signature" link in my email which then points to a signature view page back on my own server where I generate a view from the JSON data.
